cpp/src/community/k_truss_impl.cuh

```diff
@@ -0,0 +1,1164 @@
+          auto itr_lower = thrust::lower_bound(thrust::seq, dst_array_begin, dst_array_end, dst);
+          auto idx_lower =
+            thrust::distance(dst_array_begin, itr_lower);  // FIXME: remove self loops
+          auto itr_upper = thrust::upper_bound(thrust::seq, dst_array_begin, dst_array_end, dst);
+          auto idx_upper = thrust::distance(dst_array_begin, itr_upper);
+          auto dist      = thrust::distance(itr_lower, itr_upper);
+          return dist;
+        });
+      thrust::exclusive_scan(
+        handle.get_thrust_policy(), prefix_sum.begin(), prefix_sum.end(), prefix_sum.begin());
+
+      auto vertex_pair_buffer_p_q = allocate_dataframe_buffer<thrust::tuple<vertex_t, vertex_t>>(
+        prefix_sum.back_element(handle.get_stream()), handle.get_stream());
+
+      auto vertex_pair_buffer_p_r = allocate_dataframe_buffer<thrust::tuple<vertex_t, vertex_t>>(
+        prefix_sum.back_element(handle.get_stream()), handle.get_stream());
+
+      rmm::device_uvector<vertex_t> indices(num_invalid_edges, handle.get_stream());
+      thrust::tabulate(
+        handle.get_thrust_policy(), indices.begin(), indices.end(), thrust::identity<vertex_t>());
+
+      thrust::for_each(
+        handle.get_thrust_policy(),
+        indices.begin(),
+        indices.end(),
+        [invalid_first_dst           = std::get<1>(invalid_edges_buffer).begin(),
+         invalid_first_src           = std::get<0>(invalid_edges_buffer).begin(),
+         prefix_sum                  = prefix_sum.data(),
+         incoming_vertex_pairs = get_dataframe_buffer_begin(incoming_vertex_pairs),
+         vertex_pair_buffer_p_q      = get_dataframe_buffer_begin(vertex_pair_buffer_p_q),
+         vertex_pair_buffer_p_r      = get_dataframe_buffer_begin(vertex_pair_buffer_p_r),
+         num_edges                   = num_edges] __device__(auto idx) {
+          auto src             = invalid_first_src[idx];
+          auto dst             = invalid_first_dst[idx];
+          auto dst_array_begin = invalid_first_dst;
+          auto dst_array_end   = invalid_first_dst + num_edges;
+
+          auto itr_lower = thrust::lower_bound(thrust::seq, dst_array_begin, dst_array_end, dst);
+          auto idx_lower = thrust::distance(
+            dst_array_begin, itr_lower);  // Need a binary search to find the begining of the range
+
+          auto dist = prefix_sum[idx + 1] - prefix_sum[idx];
+
+          thrust::tabulate(
+            thrust::seq,
+            vertex_pair_buffer_p_q + prefix_sum[idx],
+            vertex_pair_buffer_p_q + prefix_sum[idx] + dist,
+            [incoming_vertex_pairs = incoming_vertex_pairs,
+             dst                         = dst,
+             src                         = src,
+             idx_lower                   = idx_lower](auto idx_in_segment) {
+              return thrust::make_tuple(
+                thrust::get<1>(*(incoming_vertex_pairs + idx_lower + idx_in_segment)), src);
+            });
```

Comment:
I think this can be just thrust::copy

Reply:
I tried the above suggestion and it didn't work. I will check what is going wrong.

Reply:
Nevermind